### PR TITLE
Add integration test for non-nullable files

### LINKED ISSUE #4708: File uploads - Nullability integration test
All of the current tests use a nullable Upload object, but most use-cases will probably make it non-nullable.

### PR DESCRIPTION
**Pull request by nicholascioli**
Existing integration tests for the file uploads plugin did not test for non-nullable uploads. As this is realistically the more common case, this commit adds a test to ensure that non-nullable files work as expected.

This is necessary since the router does some substitution of the variables for the file upload case which might cause validation issues if not carefully tested.

As is the case in #4754, this commit is rebased off of #4726 as certain tests fail when the output blocks.

Fixes #4708 

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [X] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -4083,9 +4083,9 @@ dependencies = [
 
 [[package]]
 name = "mio"
-version = "0.8.9"
+version = "0.8.11"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "3dce281c5e46beae905d4de1870d8b1509a9142b62eedf18b443b011ca8343d0"
+checksum = "a4a650543ca06a924e8b371db273b2756685faae30f8487da1b56505a8f78b0c"
```
Comment (IvanGoncharov) [inline Cargo.lock:4088]: Is this package used inside tests?
Or it is accidential?

File: apollo-router/src/plugins/telemetry/fmt_layer.rs [hunk 2]
```diff
@@ -167,28 +167,58 @@ where
...
-            };
-            if self.fmt_event.format_event(&ctx, &mut buf, event).is_ok() {
-                let mut writer = self.make_writer.make_writer();
-                if let Err(err) = std::io::Write::write_all(&mut writer, buf.as_bytes()) {
-                    if err.kind() != std::io::ErrorKind::WouldBlock {
-                        let _ = std::io::stderr().write_all(
-                            format!("cannot flush the logging buffer, this is a bug: {err:?}")
-                                .as_bytes(),
-                        );
+            let mut buf = &mut *(buf.borrow_mut());
```
Comment (IvanGoncharov) [inline apollo-router/src/plugins/telemetry/fmt_layer.rs:170]: Diff sims to contain unrelated change.
Can you fix this PR?

**Review by IvanGoncharov — APPROVED** [review]
Looks good, except commit list containing unrelated commits 👍

File: Cargo.lock (hunk 1 above)
Comment (nicholascioli) [reply Cargo.lock]: Not sure how this got pulled in. I've rebased and removed this change!

File: apollo-router/src/plugins/telemetry/fmt_layer.rs (hunk 2 above)
Comment (nicholascioli) [reply apollo-router/src/plugins/telemetry/fmt_layer.rs]: Again, not sure how this got here. Rebased!
